src/parser.yy

```diff
@@ -162,411 +153,542 @@
 %token L_OR                      "lor"
 %token L_NOT                     "lnot"
 
-%type <uint32_t>                         qualifiers
-%type <AstTypeIdentifier *>              type_id
-%type <AstRelationIdentifier *>          rel_id
-%type <AstType *>                        type
-%type <AstComponent *>                   component component_head component_body
-%type <AstComponentType *>               comp_type
-%type <AstComponentInit *>               comp_init
-%type <AstFunctorDeclaration *>          functor_decl
-%type <std::string>                      functor_type
-%type <std::string>                      functor_typeargs
-%type <AstRelation *>                    attributes non_empty_attributes relation_body
-%type <std::vector<AstRelation *>>       relation_list relation_decl
-%type <AstArgument *>                    arg
-%type <AstAtom *>                        arg_list non_empty_arg_list atom
-%type <std::vector<AstAtom*>>            head
-%type <RuleBody *>                       literal term disjunction conjunction body
-%type <AstClause *>                      fact
-%type <AstPragma *>                      pragma
-%type <std::vector<AstClause*>>          rule rule_def
-%type <AstExecutionOrder *>              exec_order exec_order_list
-%type <AstExecutionPlan *>               exec_plan exec_plan_list
-%type <AstRecordInit *>                  recordlist
-%type <AstUserDefinedFunctor *>          functor_list functor_args
-%type <AstRecordType *>                  recordtype
-%type <AstUnionType *>                   uniontype
-%type <std::vector<AstTypeIdentifier>>   type_params type_param_list
-%type <std::string>                      comp_override
-%type <AstIO *>                          key_value_pairs non_empty_key_value_pairs iodirective_body
-%type <std::vector<AstIO *>>             iodirective_list
-%type <std::vector<AstLoad *>>           load_head
-%type <std::vector<AstStore *>>          store_head
-
-%destructor { for (auto* cur : $$) { delete cur; } } head rule rule_def iodirective_list load_head store_head relation_list relation_decl
-%destructor { delete $$; } body conjunction disjunction literal term
-%destructor { delete $$; } rel_id
-%destructor { delete $$; } attributes non_empty_attributes
-
-%printer { yyoutput << $$; } <*>;
-
-%precedence AS
+/* -- Non-Terminal Types -- */
+%type <std::vector<std::string>>        identifier
```

Comment:
Is there an ordering to this list? I don't immediately see it. If there is not, impose one to help with readability. (lexicographical or semantic)